src/components/Modals/MembersSkillUpdateModal/Presentation.tsx

```diff
@@ -16,27 +24,102 @@ export default function MembersSkillUpdateModalPresentation({
   onClose: () => void;
   isOpen: boolean;
 }) {
+  const { username, picture, firstName, lastName } =
+    useSelector((state: RootState) => state.superUserOption);
+  const [isTagsOpen, setIsTagsOpen] = useState(false);
+  const [searchTags, setSearchTags] = useState('');
+  const { tagsWithLevel } = useGetLevels();
+  const { data, isLoading: isSkillsLoading } = useGetSkillsQuery(username);
+  const [ addNewSkill, { isLoading: isAddSkillLoading } ] = useAddNewSkillMutation();
+
+  const skills = data?.skills;
+
+  const filteredTagsData = (
+    tags: tagsWithLevelType[],
+    skills: skillsType[]
+  ) => {
+    if (searchTags !== '') {
+      return tags?.filter((tag) =>
+        tag.name.toLowerCase().includes(searchTags.toLowerCase())
+      );
+    } else if (skills?.length >= 0) {
+      return tags?.filter(
+        (tag) =>
+          !skills?.some((skill) => skill.tagId === tag.tagId && skill.levelId === tag.levelId)
+      );
+    }
+    return tags;
+  };
+
+  const filteredTags = filteredTagsData(tagsWithLevel, skills);
+
   return (
     <Modal onClose={onClose} isOpen={isOpen} isCentered>
       <ModalOverlay />
-      <ModalContent>
-        <ModalHeader>Update User Role</ModalHeader>
-        <ModalCloseButton />
-        <ModalBody
+      <ModalContent
+        sx={{
```

Comment:
please make everything in modules.css
this is not readable and we can easily fall into CSS issues which will be difficult to debug

Reply:
Sure i will do that